### PR TITLE
Add a note logging_callback only works in single process situation

### PR DESCRIPTION
**Pull request by himkt**
🔗 https://github.com/optuna/optuna/issues/3134

## Motivation
This PR adds a note to FAQ that `logging_callback` doesn't work as expected with `n_jobs>1`.

### CODE REVIEW

File: docs/source/faq.rst [hunk 1]
```diff
@@ -373,6 +373,8 @@ Here's an example:
     study = optuna.create_study()
     study.optimize(objective, n_trials=100, callbacks=[logging_callback])
 
+Note that this callback would not work as expected when you try to optimize a objective function with `n_jobs>1`.
```
Comment (himkt) [inline docs/source/faq.rst:376]: Not sure whether I should put a reason in detail here. 💭

**Review by himkt — COMMENTED** [review]
@hvy I'd like your feedback. 🙋

File: docs/source/faq.rst (hunk 1 above)
Comment (hvy) [reply docs/source/faq.rst:376]: We could elaborate a bit perhaps. Something along

```suggestion
Note that this callback may show incorrect values when you try to optimize an objective function with `n_jobs>1` (or other forms of distributed optimization) due to its reads and writes to storage which are prone to race conditions.
```

**Comment by hvy** [comment]
@keisuke-umezawa , could you also review this one?

File: docs/source/faq.rst (hunk 1 above)
Comment (nzw0301) [reply docs/source/faq.rst:376]: Nit
```suggestion
Note that this callback would not work as expected when you try to optimize a objective function with ``n_jobs!=1``.
```

Comment (himkt) [reply docs/source/faq.rst]: Thank you @hvy @nzw0301 for the suggestions! I applied your changes in 463c890 (sorry I didn't commit your suggestions, suggestions conflicted each other to let me manual apply...🔥)

**Review by hvy — APPROVED** [review]
LGTM!

**Comment by hvy** [comment]
@nzw0301 please feel free to review and/or merge if you'd like. 🙇

**Review by nzw0301 — APPROVED** [review]
LGTM

**Pull request merged**